rig/machine_control/bmp_controller.py

```diff
@@ -168,9 +168,8 @@ def _send_scp(self, cabinet, frame, board, *args, **kwargs):
 
         return connection.send_scp(length, 0, 0, board, *args, **kwargs)
 
-    @ContextMixin.use_contextual_arguments
-    def get_software_version(self, cabinet=Required, frame=Required,
-                             board=Required):
+    @ContextMixin.use_contextual_arguments()
```

Comment:
Yeah :/ this is a shame...

One way to fix it is to detect if you've received a single function as the only argument, and if so, invoke the decorator... It suddenly becomes a bit bizarre though since it breaks consistency and makes everything rather more magic than it needs to be.